dev-docs/howto/bare-metal/README.md

```diff
@@ -0,0 +1,66 @@
+# Bare-metal SNP setup for Constellation
+
+## Prepare Host
+
+The bare-metal host machine needs to be able to start SEV-SNP VMs.
+A thorough explanation can be found here: <https://github.com/AMDESE/AMDSEV/tree/snp-latest>.
+
+First checkout the snp-latest branch:
+
+```bash
+git clone https://github.com/AMDESE/AMDSEV.git
+cd AMDSEV
+git checkout snp-latest
+```
+
+Then enable TPM2 support by setting `-DTPM2_ENABLE` in the OVMF build command
+found in `common.sh`:
+
+```patch
+diff --git a/common.sh b/common.sh
+index 9eee947..52bf507 100755
+--- a/common.sh
++++ b/common.sh
+@@ -155,7 +155,7 @@ build_install_ovmf()
+                GCCVERS="GCC5"
+        fi
+
+-       BUILD_CMD="nice build -q --cmd-len=64436 -DDEBUG_ON_SERIAL_PORT=TRUE -n $(getconf _NPROCESSORS_ONLN) ${GCCVERS:+-t $GCCVERS} -a X64 -p OvmfPkg/OvmfPkgX64.dsc"
++       BUILD_CMD="nice build -q --cmd-len=64436 -DTPM2_ENABLE -DDEBUG_ON_SERIAL_PORT=TRUE -n $(getconf _NPROCESSORS_ONLN) ${GCCVERS:+-t $GCCVERS} -a X64 -p OvmfPkg/OvmfPkgX64.dsc"
+
+        # initialize git repo, or update existing remote to currently configured one
+        if [ -d ovmf ]; then
+```
+
+Build and package the binaries. Then install the newly build kernel:
+
+```bash
+./build.sh --package
+cd linux
+dpkg -i linux-image-6.9.0-rc7-snp-host-05b10142ac6a_6.9.0-rc7-g05b10142ac6a-2_amd64.deb
+```
+
+Reboot, verify that the right BIOS setting are set as described in
+<https://github.com/AMDESE/AMDSEV/tree/snp-latest?tab=readme-ov-file#prepare-host>
+and select the new kernel in the boot menu. Note that GRUB usually automatically
+select the newest installed kernel as default.
+
+Download a Constellation qemu image, the `constellation-conf.yaml`, and
+the `launch-constellation.sh` script in the directory right next to the
+`AMDSEV` folder.
+
+```bash
+wget https://raw.githubusercontent.com/edgelesssys/constellation/main/dev-docs/howto/bare-metal/launch-constellation.sh
+wget https://cdn.confidential.cloud/constellation/v1/ref/main/stream/console/v2.17.0-pre.0.20240516182331-5fb2a2cb89f2/image/csp/qemu/qemu-vtpm/image.raw
+wget < link to the constellation CLI provided by Edgeless >
+wget < link to the constellation config provided by Edgeless >
```

Comment:
```suggestion
wget https://cdn.confidential.cloud/constellation/v1/ref/main/stream/console/v2.16.4/image/csp/qemu/qemu-vtpm/image.raw
wget https://github.com/edgelesssys/constellation/releases/download/v2.16.4/constellation-linux-amd64
chmod +x constellation-linux-amd64
sudo mv constellation-linux-amd64 /usr/local/bin/constellation
constellation config generate qemu
```
Apart from the new metadata-api, this set-up doesnt require any changes to the image itself, correct?
In that case we can just use the v2.16.3 image.
Not sure if changes to the config are even necessary.

Reply:
I tried the v2.16.3 release images, but I get the following error:

Reply:
Since we haven't concluded that we want to integrate the TPM SecureBoot "fix" into mainline, we still need to build a fork for the CLI here, so no release CLI.

Reply:
Ah, this is likely due to the changes from https://github.com/edgelesssys/constellation/pull/3011
Assuming the TPM SecureBoot set up is properly fixed in the OVMF, we should be able to just use v2.17.0 release CLI and image once they are released, correct?

Reply:
Yes, as soon as v2.17.0 images are released we should be able to use them and as soon as the OVMF/TPM suff has been fixed we can omit the custom CLI and microservices.
But there's currently no effort to fix this in the OVMF, is there?

Reply:
From what I remember about our out-of-GitHub discussion, we decided that using a forked tpm library is fine for testing things out, but we should fix the problem in our OVMF if we want to actually advertise this as something people can use.
For this document its probably fine to keep it as is.